src/app/static/src/tests/surveyAndProgram/actions.test.ts

```diff
@@ -293,6 +293,33 @@ describe("Survey and programme actions", () => {
 
     });
 
+    it("it validates, commits and marks state ready", async () => {
```

Comment:
This test looks great. Would be nice to have test of error responses coming back for some / all of the requests, and that the selected datatype is upated  as expected.